### PR TITLE
Rewrite hockey_gerrit to use shenzhen

### PR DESCRIPTION
**Pull request by bootstraponline**

### CODE REVIEW

File: lib/hockey_gerrit.rb [hunk 1]
```diff
@@ -1,41 +1,91 @@
...
+      raise 'Must set HOCKEY_TOKEN' unless HOCKEY_TOKEN
+      raise 'Must provide path to IPA' unless IPA
+      raise 'IPA doesn\'t exist' unless File.exist?(IPA)
+      raise 'Must provide BUILD_URL' unless BUILD_URL
+
+      gerrit = gerrit_refspec
+      log = git_log
+      release_notes = "#{gerrit}\n#{log}"
+
+      tries = 1
```
Comment (rolyatwilson) [inline lib/hockey_gerrit.rb]: only 1 try?

File: lib/hockey_gerrit.rb [hunk 2]
```diff
@@ -1,41 +1,91 @@
...
-    changes(ENV['GERRIT_REFSPEC'])
-    assign_log
-    change_line
-    write_file
+        config_url = response.body ? response.body['config_url'] : ''
+        config_url.gsub!('https://upload.hockeyapp.net/', 'https://rink.hockeyapp.net/')
+        puts "Build uploaded to: #{config_url}"
+      rescue
+        puts "Retrying upload. #{tries} attempts remaining..."
+        retry unless (tries -= 1).zero?
```
Comment (rolyatwilson) [inline lib/hockey_gerrit.rb]: won't this only run once?

File: lib/hockey_gerrit.rb (hunk 1 above)
Comment (bootstraponline) [inline lib/hockey_gerrit.rb]: This is why we do code reviews! Nice catch. 

Fixed. I had temporarily set it to 1 for debugging.

File: lib/hockey_gerrit.rb (hunk 2 above)
Comment (bootstraponline) [inline lib/hockey_gerrit.rb]: now it'll run 5 times 😄

**Comment by rolyatwilson** [comment]
Looks great to me!

**Pull request merged**